core/src/main/java/io/grpc/internal/ManagedChannelImpl.java

```diff
@@ -1671,143 +1671,157 @@ final class NamesResolved implements Runnable {
         @SuppressWarnings("ReferenceEquality")
         @Override
         public void run() {
-          if (ManagedChannelImpl.this.nameResolver != resolver) {
-            return;
-          }
+          onResult2(resolutionResult, true);
```

Comment:
Don't pass true/false. Instead, we should call the ResolutionResultListener here with the returned status code.

Reply:
Done.